### ISSUE TITLE
Constant condition check could be annyoing

### ISSUE THREAD

**Issue by silverwind** [rhysd/actionlint#594]
For example, to temporarily disable a job, I've seen such code:

```yaml
jobs:
  foo:
    if: false # disabled because it's flaky
```

Since v1.7.9, this fails to lint. As far as I'm aware there is no other option in the YAML to disable a job temporarily. So I would suggest to re-evaluate whether this check is enabled by default.

Maybe it's also an option to ignore constant conditions that have a comment on the same line, justifying their need.

**Comment by rhysd** [comment]
Why don't you just remove the job or comment out? it is easy to toggle the job/step by `if: false`, but it's also easy to be overlooked when the job/step is larger. It's useful for debugging but I feel it should not be left in a final code.

**Comment by silverwind** [comment]
One reason might be that the repo has configured [branch protection with required checks](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#require-status-checks-before-merging). Such  protection rules would fail if a job does not exist.

In any case, we've resolved the issue now and the damage was already done, so I guess there is nothing left to do here now and reverting this is probably not an option for you.

Maybe next time consider bumping the major version of the action on expected breaking changes like this.
